### ISSUE TITLE
document.defaultView is not an instance of Window

### ISSUE THREAD

**Issue by mchalapuk** [jsdom/jsdom#1556]
``` js
if (window instanceof window.Window) {
    console.log('(:'); // browser
} else {
    console.log(';('); // jsdom
}
```

https://tonicdev.com/muroc/577e64d26d5f74130004ab40

I could fix this if it's not too much work.

**Comment by Sebmaster** [comment]
I think this might be a bug with vm / our sandbox which'd probably not be easy to fix.

**Comment by domenic** [comment]
Yeah I believe this is classic vm issues in action. This should work "from the inside" though: I believe window.eval("window instanceof Window") will give true.

We may be able to hack this in Node v6 using Symbol.hasInstance.

**Comment by muroc** [comment]
```js
var window = jsdom.jsdom().defaultView
window.eval('window instanceof Window')
```
returns false

```js
var Window = require('jsdom/lib/jsdom/browser/Window');
var window = new Window({ parsingMode: 'html' });

window instanceof window.Window
```
returns true

https://tonicdev.com/muroc/577e7ce46d5f74130004ba2a

**Comment by domenic** [comment]
Interesting. Maybe we are doing something wrong that can be fixed, but given how troublesome these issues have been, I'm betting it's either a vm bug or a VM by-design issue. I can certainly look into it to confirm.

I think we can try for a Node v6 fix though, as I said.

**Comment by domenic** [comment]
**Posted by GPT-6 Astra Medium**

Closing as fixed. `window instanceof window.Window` now returns true. The test checks that the window’s prototype is `Window.prototype`; I also confirmed the reported instanceof expression in default, outside-only, and dangerously modes.

The existing [window-prototype-chain.html](https://github.com/web-platform-tests/wpt/blob/e94af787e39d4161f01f0ef78d933f1103ee32b4/html/browsers/the-window-object/window-prototype-chain.html) WPT covers this behavior and passes on current main. It is enabled for jsdom CI by [to-run.yaml](https://github.com/jsdom/jsdom/blob/556b11fc3cc4676a22e34fa45dfa09cbd95adfb1/test/web-platform-tests/to-run.yaml#L1471), with no skip or failing expectation.
